### DISCUSSION TITLE
Typescript support documentation

### DISCUSSION THREAD

**Discussion by vikramjb** [artilleryio/artillery#2496] [Artillery]
As per this link, typescript support is present but not fully supported.

https://www.artillery.io/docs/reference/engines/playwright#example-3-use-typescript-functions-as-test-functions

Is there a document/page as to which functions in typescript are supported or not supported?

**Comment by bernardobridge** [comment]
Hey @vikramjb 👋 ,

For now this would be trial and error, although many packages should work just fine. The list is dependant on which npm packages you're using in your tests, so we can't make a list of all of them, only give known ones (like `@playwright/test` which we do list there). This is due to Typescript support bundling everything in one file, and not all packages will work properly with that. 

Very soon we will be improving Typescript support to work with `@playwright/test` by default, and providing a workaround for packages that might not work, which should work in many cases.

I'll update here when that improvement is released.

_Note: converted this into a discussion, as it's a known limitation listed in the docs, not an issue._

**Reply by vikramjb** [reply]
Hello @bernardobridge 

Thank you for the quick response. Are you planning to add a draft page? If so I can help contribute to the list as well. 🙂👍

**Reply by bernardobridge** [reply]
Hi @vikramjb, could you clarify what you mean with draft page?

If you mean the list of known packages with issues, it would be added there with the Known Limitations. However, the workaround we'll soon release should allow you to make any such packages work anyway 🤞.

If you do find out about specific packages that don't work, please do let us known and we'll make sure to test them with the improvement we have planned!

Thanks!

**Reply by vikramjb** [reply]
I meant a help page. But I think it's being taken care of. Thank you for the update 🙂👍. I will keep you posted on how my journey goes as well.

**Comment by bernardobridge** [comment]
👋  The new Artillery release ([v2.0.6](https://github.com/artilleryio/artillery/releases/tag/artillery-2.0.6)) contains improvements to the Typescript support.

Some documentation:
- https://www.artillery.io/docs/reference/test-script#preventing-bundling-of-typescript-packages
- https://www.artillery.io/docs/reference/engines/playwright#known-limitations
